src/lib.rs

```diff
@@ -241,21 +327,23 @@ impl Connection<NotConnected> {
     pub fn connect(self, nats_url: &str) -> io::Result<Connection<Connected>> {
         let conn = Connection {
             state: Authenticated {},
-            options: self.options.clone(),
+            options: self.options,
         };
-        let conn = conn.connect(nats_url)?;
-        Ok(conn)
+        conn.connect(nats_url)
     }
 }
 
 impl Connection<Authenticated> {
     fn check_port(&self, nats_url: &str) -> String {
         match nats_url.parse::<SocketAddr>() {
             Ok(_) => nats_url.to_string(),
-            Err(_) => match nats_url.find(':') {
-                Some(_) => nats_url.to_string(),
-                None => format!("{}:4222", nats_url),
-            },
+            Err(_) => {
+                if nats_url.find(':').is_some() {
+                    nats_url.to_string()
+                } else {
+                    format!("{}:4222", nats_url)
+                }
+            }
```

Comment:
A rule of thumb for when to use match or an if: use match when you need more than 2 arms, otherwise if/if-let

Reply:
Got it, thanks!